### PR TITLE
WIP draft circle 2.0

### PR DESCRIPTION
**Pull request by ashander**
per #417

### CODE REVIEW

**Comment by ashander** [comment]
Oops I messed up the yml formatting (they require some indentation that's not enforced by yml validators -_- )

Some resources:

https://circleci.com/docs/2.0/migrating-from-1-2/ https://circleci.com/docs/2.0/circleci-images/

**Comment by ashander** [comment]
Ugh. For some reason hdf5.h is not installed in the same place in debian jessie as it was in whatever deb/ubuntu platform 1.0 used. (Same issue as https://github.com/NVIDIA/DIGITS/issues/156)

Not sure what the cleanest fix is here

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            sudo apt-get install libgsl0-dev libhdf5-serial-dev libcunit1-dev libconfig-dev
+            pip install --user -r requirements/development.txt
+      - save_cache:
+          key: msprime-{{ .Branch }}
+          paths:
+            - "/home/circleci/.local"
+      - run:
+          name: Compile
+          command: |
+            CFLAGS=--coverage python setup.py build_ext --inplace
```
Comment (jeromekelleher) [inline .circleci/config.yml]: Try "CC=h5cc CFLAGS=--coverage python setup.py build_ext --inplace" here.

h5cc should know where hdf5.h and the libs are.

**Comment by jeromekelleher** [comment]
This looks great, thanks @ashander. See possible solution for hdf5 stuff above.

**Comment by ashander** [comment]
h5cc worked, eventually. after some other fixes (the high level tests pass yay) I ran into a bug with the version number in LL tests. apparently it's not being set. unsure the cause and don't have access to debug. since you own the CI account you could rebuild and SSH into the instance, though. or maybe you know what's causing this

```
======================================================================
FAIL: test_library_versions (tests.test_lowlevel.TestModule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/msprime/tests/test_lowlevel.py", line 395, in test_library_versions
    self.assertEqual(version_str, _library_version)
AssertionError: 'undefined' != '0.5.1.dev20+ng79f5a48'
- undefined
+ 0.5.1.dev20+ng79f5a48
```

**Comment by jeromekelleher** [comment]
This is great, thanks @ashander. I'll take it from here.

**Comment by ashander** [comment]
@jeromekelleher sounds good =)

**Comment by ashander** [comment]
Nice. Thanks for the fix / workaround

**Comment by jeromekelleher** [comment]
> Nice. Thanks for the fix / workaround
Yeah, not sure what's happening here. I moved discussion to #423.

Thanks for all the work getting this set up!

**Pull request merged**